laurent22/joplin
# Desktop: Fixes #8742: Show pop-up to enable safe mode after a crash

# Summary
This pull request
1. Fixes the issue that was preventing the `wasClosedSuccessfully` setting from saving on exit (related to [the issue originally raised here](https://discourse.joplinapp.org/t/pre-release-v2-8-is-now-available-updated-27-april/25158/54?u=personalizedrefriger)).
2. Re-enables the prompt to start in safe mode that appears after a crash.

This should fix #8742.

# To-do

- [ ] Run before exit logic on system shutdown/restart.

# Testing

To verify that the dialog appears as expected,
1. Launch the app from the terminal (e.g. `npx electron .`)
2. Close the app with <kbd>Ctrl</kbd>-<kbd>c</kbd>
3. Launch the app again

After step 3, a prompt to start in safe mode should appear.

To verify that [the issue that originally caused this feature to be disabled](https://discourse.joplinapp.org/t/pre-release-v2-8-is-now-available-updated-27-april/25158/50) has not returned,
1. Launch the app from the terminal **in release mode** (`npx electron .` — [the multiple instances check is not run when in dev mode](https://github.com/laurent22/joplin/blob/ddc83ec15370884fb8d06bd2cef44544cfac33e9/packages/app-desktop/ElectronAppWrapper.ts#L325)).
2. From a different terminal, attempt to launch another instance of the app (also with `npx electron .`).
3. Quit the app (test with both "file > quit" and exiting by right-clicking, then selecting "quit" from the taskbar icon).
4. Launch the app again.

After step 4, the safe mode prompt **should not** appear.

This has been tested on
- [x] Linux (Ubuntu 23.04)
- [ ] MacOS (to-do)
- [ ] Windows (to-do)
